addons/mail/models/mail_channel_rtc_session.py

```diff
@@ -95,19 +95,23 @@ def _notify_peers(self, notifications):
                 payload_by_target[target]['notifications'].append(content)
         return self.env['bus.bus']._sendmany([(target, 'mail.channel.rtc.session/peer_notification', payload) for target, payload in payload_by_target.items()])
 
-    def _mail_rtc_session_format(self, complete_info=True):
+    def _mail_rtc_session_format(self, fields=None):
         self.ensure_one()
-        vals = {
-            'id': self.id,
-            'channelMember': self.channel_member_id.mail_channel_member_format()[0],
-        }
-        if complete_info:
-            vals.update({
-                'isCameraOn': self.is_camera_on,
-                'isDeaf': self.is_deaf,
-                'isSelfMuted': self.is_muted,
-                'isScreenSharingOn': self.is_screen_sharing_on,
-            })
+        if not fields:
+            fields = {'id': True, 'channelMember': {'id': True, 'channel': {}, 'persona': {'partner': {'id', 'name', 'im_status'}, 'guest': {'id', 'name', 'im_status'}}}, 'isCameraOn': True, 'isDeaf': True, 'isSelfMuted': True, 'isScreenSharingOn': True}
+        vals = {}
+        if 'id' in fields:
+            vals['id'] = self.id
+        if 'channelMember' in fields:
+            vals['channelMember'] = self.channel_member_id._mail_channel_member_format(fields=fields.get('channelMember')).get(self.channel_member_id)
+        if 'isCameraOn' in fields:
+            vals['isCameraOn'] = self.is_camera_on
+        if 'isDeaf' in fields:
+            vals['isDeaf'] = self.is_deaf
+        if 'isSelfMuted' in fields:
+            vals['isSelfMuted'] = self.is_muted
+        if 'isScreenSharingOn' in fields:
+            vals['isScreenSharingOn'] = self.is_screen_sharing_on
```

Comment:
This look way too manual to me.
We could iterate over fields, assign the value based on the key and throw an error if the key is invalid ? `_mail_channel_member_format` could just be a condition in the loop.

Reply:
As posted above, indeed this is too manual, and we'd like an API for this, but this is way out of scope of the current task, of which the diff is already quite big.